### ISSUE TITLE
[macros] evaluate support for `avoid_renaming_method_parameters`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4891]
[avoid_renaming_method_parameters](https://dart.dev/tools/linter-rules/avoid_renaming_method_parameters) is specific to overridden methods.

Should it be extended to augmented methods/functions too?

/cc @goderbauer @munificent @lrhn @natebosch @jakemac53

**Comment by jakemac53** [comment]
Yes, definitely I think it should. All the same justifications apply.

**Comment by lrhn** [comment]
Completely generally, lints should apply to the end result of augmentation, since that's what the program really is. Something isn't missing if it's added by an augmentation.

*Some* lints can also apply to intermediate steps, like this one. There is no reason to have a declaration with a different parameter name than the superclass, and then an augmentation on top which changes back to the superclass name.

Other lints only care about the end result. Something like "always type public API" should only trigger if there are no types after augmentation.

**Comment by pq** [comment]
Augmented methods are supported w/ https://github.com/dart-lang/sdk/commit/33f94fe60aa49940e27125eeba1256d1b779d383. Will update the title to track future (possible) support for functions.
